Services/UI/OJS.Services.Ui.Business/Implementations/ParticipantsBusinessService.cs

```diff
@@ -53,8 +53,11 @@ public class ParticipantsBusinessService : IParticipantsBusinessService
         }
         else if (contest.IsOnsiteExam)
         {
-            participant.ParticipationStartTime = DateTime.Now;
-            participant.ParticipationEndTime = contest.EndTime;
+            participant.ParticipationStartTime = DateTime.UtcNow;
+            if (contest.EndTime != null)
+            {
+                participant.ParticipationEndTime = contest.EndTime;
+            }
```

Comment:
Is there a need for the `if` clause? If contest.EndTime is null, the participant.ParticipationEndTime will also be set to null.